ci/gpu/build.sh

```diff
@@ -80,6 +80,14 @@ pip install "git+https://github.com/dask/distributed.git"
 logger "conda install feather-format"
 conda install "feather-format" "cupy>=6.0.0"
 
+# set environment variable for numpy 1.16
+# will be enabled for later versions by default
+np_ver=$(python -c "import numpy; print('.'.join(numpy.__version__.split('.')[:-1]))")
+if [ "$np_ver" == "1.16" ];then
+  logger "export NUMPY_EXPERIMENTAL_ARRAY_FUNCTION=1"
+  export NUMPY_EXPERIMENTAL_ARRAY_FUNCTION=1
+fi
```

Comment:
We should make sure that gpuci-build-environment has numpy >= 1.16, I believe it does but would be good to explicitly version it.

Reply:
Should be fixed with : 

https://github.com/rapidsai/cudf/pull/2157/files#diff-e19d03ed8e05824bbc937b07c4fddd6fR45

The explicit check for version is because in `numpy 1.17` we don't need to set it explicitly. This environment variable is supposed to be for this one single numpy version.  

From `nep-18`:

https://www.numpy.org/neps/nep-0018-array-function-protocol.html
```
Note
Dispatch with the __array_function__ protocol has been implemented but is not yet enabled by default:

In NumPy 1.16, you need to set the environment variable NUMPY_EXPERIMENTAL_ARRAY_FUNCTION=1 before importing NumPy to test NumPy function overrides.
In NumPy 1.17, the protocol will be enabled by default, but can be disabled with NUMPY_EXPERIMENTAL_ARRAY_FUNCTION=0.
Eventually, expect to __array_function__ to always be enabled.
```